### ISSUE TITLE
Can't build build mt76 package with CMAKE_GENERATOR=Ninja on host.

### ISSUE THREAD

**Issue by drizt** [openwrt/openwrt#16164]
### Describe the bug

mt76 explicity requires to not use ninja `PKG_USE_NINJA:=0`. But if in host environment `CMAKE_GENERATOR=Ninja` ninja-files will be generated by default. To prevent this need to explicity set generator for Makefile in cmake.mk.

### OpenWrt version

r24012-d8dd03c46f

### OpenWrt release

23.05.4

### OpenWrt target/subtarget

ramips/mt7621

### Device

ASUS RT-AX53U

### Image kind

Self-built image

### Steps to reproduce

export CMAKE_GENERATOR=Ninja
... setup menuconfig for ramips/mt7621
make -j $(nproc) V=sc package/kernel/mt76/compile

### Actual behaviour

```
make -j $(nproc) V=sc package/kernel/mt76/compile
...
mipsel-openwrt-linux-musl-ld -r -m elf32ltsmip -z noexecstack --no-warn-rwx-segments --build-id=sha1  -T scripts/module.lds -o /home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/mt792x-lib.ko /home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/mt792x-lib.o /home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/mt792x-lib.mod.o;  make -f ./arch/mips/Makefile.postlink /home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/mt792x-lib.ko
make -C /home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/tools
make[3]: Entering directory '/home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/tools'
make[3]: *** No targets specified and no makefile found.  Stop.
make[3]: Leaving directory '/home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/tools'
make[2]: *** [Makefile:639: /home/taurus/develop/openwrt/build_dir/target-mipsel_24kc_musl/linux-ramips_mt7621/mt76-2024-04-03-1e336a85/.built] Error 2
```

### Expected behaviour

Successfull package compilation.

### Additional info

Patch to fix this.

``` diff
diff --git a/include/cmake.mk b/include/cmake.mk
index 9b169b0693..278d6e29e0 100644
--- a/include/cmake.mk
+++ b/include/cmake.mk
@@ -68,6 +68,8 @@ ifeq ($(HOST_USE_NINJA),1)
   define Host/Uninstall/Default
    +$(NINJA) -C $(HOST_CMAKE_BINARY_DIR) uninstall
   endef
+else
+  CMAKE_HOST_OPTIONS += -DCMAKE_GENERATOR="Unix Makefiles"
 endif
 
 ifeq ($(PKG_USE_NINJA),1)
@@ -80,6 +82,8 @@ ifeq ($(PKG_USE_NINJA),1)
   define Build/Install/Default
    +DESTDIR="$(PKG_INSTALL_DIR)" $(NINJA) -C $(CMAKE_BINARY_DIR) install
   endef
+else
+  CMAKE_OPTIONS += -DCMAKE_GENERATOR="Unix Makefiles"
 endif
 
 define Build/Configure/Default
```

### Terms

- [X] I am reporting an issue for OpenWrt, not an unsupported fork.

**Comment by drizt** [comment]
Fixed in 4646aa169986036772b9f75393c08508d20ddf8b.

**Comment by drizt** [comment]
Fixed in 4646aa169986036772b9f75393c08508d20ddf8b.
